### ISSUE TITLE
Npm reference broken

### ISSUE THREAD

**Issue by pyladune** [OvermindDL1/polymer-brunch#1]
seems it is not available on npm

**Comment by OvermindDL1** [comment]
Ah indeed it is not, I never really thought that anyone else might ever use this considering polymer's ecosystem is very webpack/brunch (yick on both) oriented, and this was just a quick hack to get my polymer stuff working in brunch (although I'm surprised at how well it has worked for me).  :-)

I have it in my dependencies via:
```json
    "polymer-brunch": "github:OvermindDL1/polymer-brunch",
```

I can add it to npm if you want though?  :-)

**Comment by pyladune** [comment]
Thanks for feedback !

i am VERY surprised that nearly no one use polymer with phoenix. I'm
starting so perhaps i do not see many drawbacks !

The 2 i found for now are the tooooons of html tags that are generated
(but it works !!) and the number of request at load of the pages. It is
for this last point looked for your polymer-brunch. i expect that
vulcanize them will speed up the process...

did you used polymer + phoenix/brunch in some project ?

My concerns is also progressive apps

Thanks

Seb

**Comment by OvermindDL1** [comment]
> i am VERY surprised that nearly no one use polymer with phoenix. I'mstarting so perhaps i do not see many drawbacks !

The only issues I've really had are just the ecosystem of it is rather poorly designed (the tools) but the polymer webcomponents themselves are fantastic!  Plus they use bower, so I just dump the modules into my project straight so I don't have to touch that bower crap (it really needs to die already, npm can do what it does now, but better).

> The 2 i found for now are the tooooons of html tags that are generated (but it works !!) and the number of request at load of the pages. It is for this last point looked for your polymer-brunch. i expect that vulcanize them will speed up the process...

If you are curious as to how I use it is that I have a `global.polymer` file that imports all of the 'globally accessible stuff', which brunch will combine that output with the main app.js from phoenix.  I have a few other `*.polymer` files for individual pages that only reference code that those sole pages use.  My main site loads the app.js and so forth and I built a dispatching router so that based on the path accessed it queries phoenix over a websocket of what should be shown, phoenix responds back with information of what element to render and what things to import first if any, this way the client just downloads the code of what they access and they cache it for later access as well, it makes the pages feel *instant* after that first load.  :-)

> did you used polymer + phoenix/brunch in some project ?

Extensively, I've talked about it a bit on the Elixir/Phoenix forums at:  https://elixirforum.com/

> My concerns is also progressive apps

Although the webcomponent spec (that polymer uses) originally planned for progressive apps, you can blame Apple for stopping it, the progressive aspect of webcomponents (the `is` attribute) is looking to not be supported because of Apple's stupidity, so if you want progressive then you want to make a traditional fallback.  I'm not needing to worry about that since my site is fully internal to my company.  :-)

_/me really hates what Apple is doing to the webcomponent spec, trying to damage it..._

**Comment by pyladune** [comment]
do you have some references about apple thwarting web components ?

this link seems to show they embeed that in safari 10 :
https://developer.apple.com/library/prerelease/content/releasenotes/General/WhatsNewInSafari/Articles/Safari_10_0.html

also any minimal project with polymer brunch hooked ?

Thanks

seb

**Comment by OvermindDL1** [comment]
> do you have some references about apple thwarting web components ?

They are thwarting not webcomponents itself, but the ability for them to be progressive via the `is` attribute.  :-)

See the discussion at:  https://github.com/w3c/webcomponents/issues/509

**Comment by pyladune** [comment]
ok,

Does it measn that if I do not extend any of the native elements of
polymer, then i do not have limitations on IOS ?

**Comment by pyladune** [comment]
also one limitation i have with polymer is use of select tag on mobile
devices.

While a normal select in HTML is triggering native features on mobile
that are pretty convenient for use, polymer like select box (drop-down
menu + list) do not. So for now i use normla Html select. Any advice on
that ?

Cheers

**Comment by OvermindDL1** [comment]
> Does it measn that if I do not extend any of the native elements of polymer, then i do not have limitations on IOS ?

Yeah, if you do not need progressive rendering then there is no issue.

However, Polymer still polyfills `is` anyway, so you can still do progressive rendering if you use polymer, just not if you use 'raw' webcomponents.

>also one limitation i have with polymer is use of select tag on mobile devices. While a normal select in HTML is triggering native features on mobile that are pretty convenient for use, polymer like select box (drop-down menu + list) do not. So for now i use normla Html select. Any advice on that ?

A drop-down menu + list is not a select box, and you can swap between them based on the browser too, that is what the `if` template is for.
